.github/workflows/checker.c

```diff
@@ -1,21 +1,30 @@
 #include <stdio.h>
 #include <assert.h>
-
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
+int isTemperatureOk(float temperature) {
+  if (temperature < 0 || temperature > 45) {
     printf("Temperature out of range!\n");
     return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
+  }
+  return 1;
+}
+ int isSocOk(float soc) {
+   if (soc < 20 || soc > 80) {
+     printf("State of Charge out of range!\n");
+     return 0;
+   }
+   return 1;
+}
+int isChargeRateOk(float chargeRate) {
+  if (chargeRate > 0.8) {
     printf("Charge Rate out of range!\n");
     return 0;
   }
   return 1;
 }
-
+int batteryIsOk(float temperature, float soc, float chargeRate) {
+  return isTemperatureOk(temperature) && isSocOk(soc) && isChargeRateOk(chargeRate);
```

Comment:
if/else range check is repeated in every parameter check function ! DRY principle violation